File: backend/src/modules/license/schemas.py
"""License 管理模块 - Pydantic Schema。
v1.1 变更：
- P1-7：LicenseStatusOut 移除 machine_code / current_machine_code，
        严格遵循《API接口文档》§ 8.1 契约。
- P2-3：删除未使用的 LicensePayload / LicenseFileContent。
- P2-7：LicenseActivateReq.activation_code 增加 max_length=255。
"""
from datetime import datetime
from typing import List, Optional
from pydantic import BaseModel, Field
class LicenseStatusOut(BaseModel):
    """
    License 状态响应（严格遵循《API接口文档》§ 8.1）。
    v1.1 起不再包含 machine_code / current_machine_code 字段。
    机器码信息仅用于服务端日志排障。
    """
    is_valid: bool = Field(..., description="当前 License 是否有效")
    license_type: Optional[str] = Field(None, description="授权类型")
    expires_at: Optional[datetime] = Field(None, description="过期时间")
    days_remaining: Optional[int] = Field(None, description="剩余天数")
    max_users: Optional[int] = Field(None, description="最大用户数（NULL 表示不限制）")
    current_users: int = Field(..., description="当前用户数")
    authorized_modules: List[str] = Field(
        default_factory=list, description="授权模块 ID 列表"
    )
    is_expired: bool = Field(False, description="是否已过期")
    is_expiring_soon: bool = Field(False, description="是否即将过期（≤7 天）")
class LicenseActivateReq(BaseModel):
    """在线激活请求。"""
    activation_code: str = Field(
        ...,
        min_length=1,
        max_length=255,
        description="激活码（1-255 位）",
    )
    machine_code: str = Field(
        ...,
        min_length=1,
        max_length=255,
        description="机器码（1-255 位）",
    )
class ModuleAuthorizationOut(BaseModel):
    """模块授权状态。"""
    module_id: str
    module_name: str
    is_authorized: bool
    expires_at: Optional[datetime] = None

File: backend-go/internal/modules/license/schemas.go
package license
import "time"
// LicenseStatusOut License 状态响应。
//
// 严格对应 Python 版 LicenseStatusOut：不含 machine_code / current_machine_code
// 字段（v1.1 起已移除），机器码信息仅用于服务端日志排障。
type LicenseStatusOut struct {
	IsValid           bool     `json:"is_valid"`
	LicenseType       *string  `json:"license_type"`
	ExpiresAt         *string  `json:"expires_at"`
	DaysRemaining     *int     `json:"days_remaining"`
	MaxUsers          *int     `json:"max_users"`
	CurrentUsers      int      `json:"current_users"`
	AuthorizedModules []string `json:"authorized_modules"`
	IsExpired         bool     `json:"is_expired"`
	IsExpiringSoon    bool     `json:"is_expiring_soon"`
}
// LicenseImportResp License 导入/激活响应。
type LicenseImportResp struct {
	ID                uint     `json:"id"`
	LicenseKey        string   `json:"license_key"`
	LicenseType       string   `json:"license_type"`
	ExpiresAt         string   `json:"expires_at"`
	AuthorizedModules []string `json:"authorized_modules"`
	MaxUsers          *int     `json:"max_users"`
	MachineCode       *string  `json:"machine_code"`
}
// LicenseActivateReq 在线激活请求。
type LicenseActivateReq struct {
	ActivationCode string `json:"activation_code" binding:"required,min=1,max=255"`
	MachineCode    string `json:"machine_code" binding:"required,min=1,max=255"`
}
// ModuleAuthorizationOut 模块授权状态。
type ModuleAuthorizationOut struct {
	ModuleID     string  `json:"module_id"`
	ModuleName   string  `json:"module_name"`
	IsAuthorized bool    `json:"is_authorized"`
	ExpiresAt    *string `json:"expires_at"`
}
// formatDateTime 将时间格式化为 Python datetime.isoformat() 兼容的字符串。
//
// 规则：
//   - 零值返回空串；
//   - 微秒为 0 时省略小数部分（与 Python 默认行为一致）；
//   - 微秒非 0 时保留 6 位（与 Python 的微秒精度一致）。
func formatDateTime(t time.Time) string {
	if t.IsZero() {
		return ""
	}
	t = t.UTC()
	if t.Nanosecond()/1000 == 0 {
		return t.Format("2006-01-02T15:04:05")
	}
	return t.Format("2006-01-02T15:04:05.000000")
}
